src/pybind/mgr/dashboard/frontend/src/app/shared/components/notifications-sidebar/notifications-sidebar.component.ts

```diff
@@ -83,15 +83,8 @@ export class NotificationsSidebarComponent implements OnInit, OnDestroy {
     this.last_task = window.localStorage.getItem('last_task');
 
     const permissions = this.authStorageService.getPermissions();
-    if (permissions.prometheus.read && permissions.configOpt.read) {
-      this.triggerPrometheusAlerts();
-      this.ngZone.runOutsideAngular(() => {
-        this.interval = window.setInterval(() => {
-          this.ngZone.run(() => {
-            this.triggerPrometheusAlerts();
-          });
-        }, 5000);
-      });
+    if (this.isSidebarOpened && permissions.prometheus.read && permissions.configOpt.read) {
```

Comment:
don't we need to run this in the background as well (even if side bar is closed?). to monitor any incoming alerts?

Reply:
yeah we need a global state, it was kind of scattered with subscriptions hard to trace now.
Thanks